variable/util.cpp

```diff
@@ -48,4 +50,25 @@ Variable variances(const VariableConstView &x) {
   return transform(x, element::variances);
 }
 
+/// Return true if variable values are sorted along given dim.
+///
+/// If `order` is SortOrder::Ascending, checks if values are non-decreasing.
+/// If `order` is SortOrder::Descending, checks if values are non-increasing.
+bool is_sorted(const VariableConstView &x, const Dim dim,
+               const SortOrder order) {
+  const auto size = x.dims()[dim];
+  if (size < 2)
+    return true;
+  auto out = makeVariable<bool>(Values{true});
+  if (order == SortOrder::Ascending)
+    accumulate_in_place(out, x.slice({dim, 0, size - 1}),
+                        x.slice({dim, 1, size}),
+                        core::element::is_sorted_nondescending);
+  else
```

Comment:
minor point: anything not `SortOrder::Ascending` is treated as descending. Could you replace enum with bool, or if you think that a future enum option might be added, it would be safer to `else if` and then allow `else` to error.

Reply:
I think it is commonly suggested to avoid `bool` arguments since there lead to bad readability? I do *not* think that further options would be added. Can you think of any?

Reply:
No not for `SortOrder`